src/main/java/org/apache/pulsar/io/jcloud/format/ParquetFormat.java

```diff
@@ -263,8 +268,26 @@ public ByteBuffer recordWriterBuf(Iterator<Record<GenericRecord>> records) throw
                         parquetWriter.write(protoRecord);
                     }
                 } else {
-                    org.apache.avro.generic.GenericRecord writeRecord = AvroRecordUtil
-                            .convertGenericRecord(genericRecord, rootAvroSchema);
+                    org.apache.avro.generic.GenericRecord writeRecord;
+                    if (rootAvroSchema != null && genericRecord.getSchemaType() == SchemaType.KEY_VALUE) {
+                        KeyValue<GenericRecord, GenericRecord> keyValue =
+                                (KeyValue<GenericRecord, GenericRecord>) genericRecord.getNativeObject();
+                        writeRecord = new GenericData.Record(rootAvroSchema);
+                        GenericRecord keyObject = keyValue.getKey();
+                        if (keyObject != null) {
+                            Schema keySchema = rootAvroSchema.getField("key").schema();
+                            writeRecord.put("key", AvroRecordUtil.convertGenericRecord(keyObject, keySchema));
```

Comment:
ideally for users it would be better to have an option to merge the key value and not have this level with "key" and "value"
but we can wait for the user feedback